scripts/commands/update-translations.js

```diff
@@ -0,0 +1,170 @@
+		error( 'Invalid language data format.' );
+		process.exit( 1 );
+	}
+}
+
+async function updateTranslations() {
+	const git = gitFactory();
+
+	try {
+		info( 'Executing wp i18n make-pot command...' );
+
+		const makePotResult = await execPromise(
+			`wp i18n make-pot . ${ POT_FILE_PATH } --ignore-domain`
+		);
+
+		if ( makePotResult.stderr ) {
+			error( `wp i18n make-pot failure: ${ makePotResult.stderr }` );
+			process.exit( 1 );
+		}
+
+		success( 'wp i18n make-pot command executed successfully.' );
+
+		const status = await git.status();
+
+		const isPotFileUpdated =
+			status.modified.includes( POT_FILE_PATH ) ||
+			status.created.includes( POT_FILE_PATH );
+
+		if ( ! isPotFileUpdated ) {
+			warning(
+				`${ POT_FILE_PATH } is up to date. No need to update translations.`
+			);
+			process.exit( 0 );
+		}
+
+		info( 'Retrieving language data from translate.wordpress.com...' );
+		const langData = await new Promise( ( resolve, reject ) => {
+			https
+				.get(
+					'https://translate.wordpress.com/api/projects/woocommerce/wc-calypso-bridge',
+					( response ) => {
+						let data = '';
+						response.on( 'data', ( chunk ) => {
+							data += chunk;
+						} );
+						response.on( 'end', () => {
+							resolve( data );
+						} );
+					}
+				)
+				.on( 'error', ( err ) => {
+					reject( err );
+				} );
+		} );
+
+		verifyLangData( langData );
+
+		success( 'Language data retrieved successfully.' );
+
+		const { translation_sets: translationSets } = JSON.parse( langData );
+
+		const filteredSets = translationSets.filter(
+			( set ) => set.percent_translated > REQUIRED_TRANSLATION_PERCENTAGE
+		);
+
+		for ( const lang of filteredSets ) {
+			const {
+				name: LANG_NAME,
+				locale: LANG_LOCALE,
+				slug: LANG_SLUG,
+				wp_locale: LANG_WP_LOCALE,
+			} = lang;
+
+			if (
+				LANG_NAME === undefined ||
+				LANG_LOCALE === undefined ||
+				LANG_SLUG === undefined ||
+				LANG_WP_LOCALE === undefined
+			) {
+				error(
+					'Missing required properties in language data. Skipping...'
+				);
+				continue;
+			}
+
+			const LANG_FILENAME = `languages/wc-calypso-bridge-${
+				LANG_WP_LOCALE || LANG_LOCALE
+			}.po`;
+			const poUrl = `https://translate.wordpress.com/projects/woocommerce/wc-calypso-bridge/${ LANG_LOCALE }/${ LANG_SLUG }/export-translations/?format=po`;
+
+			info(
+				`Downloading ${ LANG_NAME } (${ LANG_LOCALE }/${ LANG_SLUG } → ${
+					LANG_WP_LOCALE || LANG_LOCALE
+				})...`
+			);
+
+			try {
+				await downloadTranslations( poUrl, LANG_FILENAME );
+				success( `${ LANG_FILENAME } downloaded successfully.` );
+
+				await execPromise(
+					`msgfmt ${ LANG_FILENAME } -o ${ LANG_FILENAME.replace(
+						'.po',
+						'.mo'
+					) }`
+				);
+				success( `${ LANG_FILENAME } compiled successfully.` );
+			} catch ( err ) {
+				error(
+					`Error downloading or compiling ${ LANG_FILENAME }: ${ err.message }`
+				);
+				process.exit( 1 );
+			}
+		}
+
+		info( 'Staging language files...' );
```

Comment:
It could happen that there won't be any changes here so adding/committing would fail. It would be better to do something like:
```
if (isPotFileUpdated) {
    info ( 'Committing new POT file...' );
    commit the file
}
if (otherLanguageFilesUpdated) {
    info ( 'Committing new translation files...' );
    commit the files
}
```

Reply:
Would `npm run makepot` be executed as part of a release process, or should anyone be able to run it? If it's the latter, then committing to the current branch might not be the best idea. Perhaps this script shouldn't commit anything at all and it'd be done in tag.js instead?

Reply:
I believe the goal is to make `npm run makepot` be part of the release process. @markbiek ?